src/Program.cs

```diff
@@ -57,375 +63,468 @@ public static void Main(string[] args)
             {
                 var shouldShowHelp = false;
 
-                // command line options configuration
+                // Shutdown token sources.
+                ShutdownTokenSource = new CancellationTokenSource();
+
+                // detect the runtime environment. either we run standalone (native or containerized) or as IoT Edge module (containerized)
+                // check if we have an environment variable containing an IoT Edge connectionstring, we run as IoT Edge module
+                if (IsIotEdgeModule)
+                {
+                    WriteLine("IoTEdge detected.");
+                }
+
+                // command line options
                 Mono.Options.OptionSet options = new Mono.Options.OptionSet {
-                    // Publishing configuration options
-                    { "pf|publishfile=", $"the filename to configure the nodes to publish.\nDefault: '{PublisherNodeConfigurationFilename}'", (string p) => PublisherNodeConfigurationFilename = p },
-                    { "tc|telemetryconfigfile=", $"the filename to configure the ingested telemetry\nDefault: '{PublisherTelemetryConfigurationFilename}'", (string p) => PublisherTelemetryConfigurationFilename = p },
-                    { "sd|shopfloordomain=", $"the domain of the shopfloor. if specified this domain is appended (delimited by a ':' to the 'ApplicationURI' property when telemetry is sent to IoTHub.\n" +
```

Comment:
I guess this is not used, otherwise you have always provided legacy switches.

Reply:
i think no one has used it out there. but i will add it back.